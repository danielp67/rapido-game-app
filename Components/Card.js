import React from 'react';
import {StyleSheet} from "react-native";

const Card = (props) => {
    const {setCardOnDbClick, slotName, card, realPlayer} = props


    const doubleClick = (e) => {
        setCardOnDbClick({card:card, slotName:slotName})
    };

    let className=""
    if(slotName==="reserveSlot")
    {
        className="border border-dark border-2"
    }
    if(slotName==="rapidoSlot")
    {
        className="border border-secondary border-2"
    }


    if(realPlayer) {
        return (
            <div
                style={styles(card).playerCard}
                onTouchEndCapture={(e) => doubleClick(e)}
            >
                <p className="my-auto fs-4 fw-bold">{card.value}</p>
            </div>

        )
    }
    else{
        return (
            <div
                style={styles(card).autoPlayerCard}
            >
                <p className="my-auto">{card.value}</p>
            </div>

        )
    }

}


const styles = (props) => StyleSheet.create({

    autoPlayerCard:{
            width: '30px',
            height: '50px',
        flex:1,
        backgroundColor: props.suit,
        border:'0.1rem solid',
        borderRadius:'0.3rem',
        },

    playerCard:{
        width: '50px',
        height: '80px',
        flex:1,
        backgroundColor: props.suit,
        border:'0.1rem solid',
        borderRadius:'0.3rem',
    }
    });


export default Card;