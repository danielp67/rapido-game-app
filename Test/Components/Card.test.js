import React from 'react';
import {create, act} from 'react-test-renderer';
import Card from '../../Components/Card';

test('card', () => {
    const deck = [{card: 't', suit: 't'}];
    const mockFn = jest.fn()
    const tree = create(
        <Card
            card={deck}
            realPlayer={true}
            slotName={"test"}
            setCurrentCard={mockFn}
            setCardOnDbClick={mockFn}
            drop={mockFn}
        />
    );
    expect(tree.toJSON()).toMatchSnapshot();

    const card = tree.root.findByProps({testID : 'card'}).props
    act(() => card.onTouchEndCapture())
})

/*
test('card touch', () => {
    const tree = create(<Card />);
    const card = tree.root.findByProps({testID : 'card'}).props
    act(() => card.onTouchEndCapture())
    console.log(card)
});*/
