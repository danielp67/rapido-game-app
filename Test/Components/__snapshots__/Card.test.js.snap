// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card 1`] = `
<View
  onDoubleClick={[Function]}
  onTouchEndCapture={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": undefined,
      "border": "solid",
      "borderRadius": 3,
      "height": 80,
      "justifyContent": "center",
      "margin": 5,
      "width": 50,
    }
  }
  testID="card"
>
  <Text
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "center",
      }
    }
  />
</View>
`;
